src/Synapse/Db/DbServiceProvider.php

```diff
@@ -29,6 +29,14 @@ public function register(Application $app)
             return new Transaction($app['db']);
         });
 
+        $app->initializer(
+            'Synapse\\Db\\TransactionAwareInterface',
+            function ($object, $app) {
```

Comment:
That's not necessary. By definition, only TransactionAwareInterface implementing objects will be passed to that function.